core/services/pipeline/graph.go

```diff
@@ -90,6 +92,28 @@ func (g TaskDAG) TasksInDependencyOrder() ([]Task, error) {
 	return tasks, nil
 }
 
+func (g TaskDAG) TasksInDependencyOrderWithResultTask() ([]Task, error) {
+	tasks, err := g.TasksInDependencyOrder()
+	if err != nil {
+		return nil, err
+	}
+	// Create the final result task that collects the answers from the pipeline's
+	// outputs.  This is a Postgres-related performance optimization.
```

Comment:
It needs to go. It's no longer necessary

Reply:
Yeah I figured I'll pick it up in a follow up, seemed too big to include in this refactor

Reply:
Agreed